bait-files/src/main/java/com/google/LoginFilter.java

```diff
@@ -71,20 +72,41 @@ else if (!userIsRegistered(userService.getCurrentUser().getUserId())
         RequestDispatcher requestDispatcher = request.getRequestDispatcher("WEB-INF/profile.jsp");
         requestDispatcher.forward(request, response);
         return;
+      } else if (request.getRequestURI().endsWith("requests.html")) {
+          System.out.println("Here");
+          if (getCurrentUserPermission() == Permission.ADMIN) {
+              chain.doFilter(req, res);
+          } else {
+            RequestDispatcher requestDispatcher = request.getRequestDispatcher("WEB-INF/profile.jsp");
+            requestDispatcher.forward(request, response);  
+          }
       } else {
+        System.out.println(request.getRequestURI());
         // Case: User is logged in and registered and wants to access site resource
         chain.doFilter(req, res);
         return;
       }
     }
   }
-  private boolean userIsRegistered(String id) {
+
+  private boolean isRegistered(String id) {
     List<User> allUsers = ObjectifyService.ofy().load().type(User.class).list();
 
-    List<User> listOfUsersWithId =
+    List<User> isRegistered =
         allUsers.stream().filter(user -> user.getUserId().equals(id)).collect(toList());
 
-    return listOfUsersWithId.size() >= 1;
+    return isRegistered.size() >= 1;
+  }
+
+  private Permission getCurrentUserPermission() {
```

Comment:
Comment for what this is doing please.

Reply:
Added comment

Reply:
I meant.. more specifically. Why is it setting userPermission, just to immediately override it? Suggest something like...

```
return allUsers.stream()
   .filter(user -> 
      user.getUserId().equals(UserServiceFactory.getUserService().getCurrentUser().getUserId()))
  .collect(onlyElement())
  .getPermission();
```

Reply:
Got it!